tutorials/datasets_tutorial.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+import piton
+import piton.datasets
+import contextlib
+import datetime
+import os
+
+# This tutorial covers the two major dataset types in piton, EventCollection
+# PatientCollection
+
+target_directory = "dataset_tutorial_target"
+
+if os.path.exists(target_directory):
+    print("This demo expects an empty target directory")
+    exit(-1)
+else:
+    os.mkdir(target_directory)
+
+
+#####################################
+##### Part 1: Create events #########
+#####################################
+
+# Piton stores events within EventCollections, which are an unordered series of events
+# Note that there are three types of events, numeric, binary, and None
+
+dummy_events = [
+    piton.Event(
+        start=datetime.datetime(1995, 1, 3),
+        code=0,
+        value=float(34),
+    ),
+    piton.Event(
+        start=datetime.datetime(2010, 1, 3),
+        code=1,
+        value=memoryview(b"test_value"),
+    ),
+    piton.Event(
+        start=datetime.datetime(2010, 1, 5),
+        code=2,
+        value=None,
+    ),
+]
+
+events = piton.datasets.EventCollection(
+    os.path.join(target_directory, "events")
+)
+
+# Once we create an events object we can just start writing
+with contextlib.closing(events.create_writer()) as writer:
+    for event in dummy_events:
+        writer.add_event(patient_id=30, event=event)
```

Comment:
This feels like an important point -- that this appends events to an already existing file (so the user will need to explicitly delete the file if they want to start over)